benchmark/output_formatter.cpp

```diff
@@ -0,0 +1,181 @@
+// Unless explicitly stated otherwise all files in this repository are
+// dual-licensed under the Apache-2.0 License or BSD-3-Clause License.
+//
+// This product includes software developed at Datadog
+// (https://www.datadoghq.com/). Copyright 2022 Datadog, Inc.
+
+#include "output_formatter.hpp"
+#include "random.hpp"
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+
+namespace ddwaf::benchmark {
+
+namespace {
+
+using output_fn_type = void (*)(std::ostream &, const settings &,
+    const std::map<std::string, std::vector<runner::test_result>> &);
+
+constexpr double MILLI = 1e3;
+constexpr double MICRO = 1e6;
+
+// NOLINTBEGIN(*-narrowing-conversions,*-magic-numbers)
+double percentile(const std::vector<uint64_t> &values, unsigned percentile)
+{
+    std::size_t size = values.size();
+    std::size_t index = ceil((size * percentile) / 100.0);
+    if (index > 0) {
+        index = index - 1;
+    }
+    return values[index];
+}
+
+double standard_deviation(const std::vector<uint64_t> &values, double mean)
+{
+    double sd = 0.0;
+    for (auto v : values) { sd += (v - mean) * (v - mean); }
+    return sqrt(sd / values.size());
+}
+
+double mean(const std::vector<uint64_t> &values)
+{
+    double m = 0.0;
+    for (auto v : values) { m += v; }
+    return m / values.size();
+}
```

Comment:
I was more tempted to remove the human readable statistics altogether, rather than spend any more time on them... the same applies to the comment below, I might do that before merging.

The main reason is that the `benchmark_analyzer` tool generates a very complete report from the json (cbmf) output.